src/bootstrap/builder.rs

```diff
@@ -1657,21 +1654,27 @@ impl<'a> Builder<'a> {
         if let Some(host_linker) = self.linker(compiler.host) {
             cargo.env("RUSTC_HOST_LINKER", host_linker);
         }
-        if self.is_fuse_ld_lld(compiler.host) {
-            cargo.env("RUSTC_HOST_FUSE_LD_LLD", "1");
-            cargo.env("RUSTDOC_FUSE_LD_LLD", "1");
+
+        if !self.is_lld_direct_linker(compiler.host) {
+            match self.config.lld_mode {
+                LldMode::External => {
+                    cargo.env("RUSTC_HOST_LLD", "external");
```

Comment:
```suggestion
                    cargo.env("RUSTC_HOST_LLD", "external")
```
Nit: formatting.
Here and in src/bootstrap/test.rs too.